### PR TITLE
Added a binomial distribution formula calculator algorithm

### PR DESCRIPTION
**Pull request by XxSamixX123**
### **Describe your change:**
Added a binomial distribution algorithm
(also added in the same file a factorial function that is being used
by the binomial distribution function)

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/binomial_distribution.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+"""For more information about the Binomial Distribution -
+    https://en.wikipedia.org/wiki/Binomial_distribution"""
+
+
+def binomial_distribution(k, n, p) -> float:
```
Comment (cclauss) [inline maths/binomial_distribution.py:5]: Single letter variable names are old school.  Please replace with self-documenting variable names.

Type hints?, doctests?, Please read CONTRIBUTING.md.

**Comment by cclauss** [comment]
Codespell says: ./maths/binomial_distribution.py:14: succes ==> success

File: maths/binomial_distribution.py [hunk 2]
```diff
@@ -22,22 +17,20 @@ def binomial_distribution(k, n, p) -> float:
...
-    if n < 0 or k < 0 or type(k) != int or type(n) != int:
-        raise ValueError("the function is defined for non-negative integers k and n")
-    if p > 1 or p < 0:
-        raise ValueError("p - Probability has to be in range of 1 - 0")
-    probability = (p**k)*(1-p)**(n-k)
+    if successes > trials:
+        raise ValueError("""successes must be lower or equal to trials""")
+    if trials < 0 or successes < 0:
+        raise ValueError("the function is defined for non-negative integers")
+    if type(successes) != int or type(trials) != int:
```
Comment (cclauss) [inline maths/binomial_distribution.py:25]:
```suggestion
    if not isinstance(successes, int) or not isinstance(trials, int):
```
As discussed in PEP8

File: maths/binomial_distribution.py [hunk 3]
```diff
@@ -61,6 +54,5 @@ def factorial(n) -> int:
     return result
 
 if __name__ == "__main__":
-    print ("Probability of 2 successes out of 4 trails")
-    print ("with probability of 0.75 is : ")
-    print (str(binomial_distribution(2, 4, 0.75)))
```
Comment (cclauss) [inline maths/binomial_distribution.py]: You can leave these three lines in place if you want.

**Comment by cclauss** [comment]
Travis CI says...

$ __flake8 --ignore=E203,W503 --max-complexity=25 --max-line-length=88 --statistics --count .__
```
./maths/binomial_distribution.py:29:36: E226 missing whitespace around arithmetic operator
./maths/binomial_distribution.py:29:39: E226 missing whitespace around arithmetic operator
./maths/binomial_distribution.py:29:54: E226 missing whitespace around arithmetic operator
./maths/binomial_distribution.py:33:41: E226 missing whitespace around arithmetic operator
./maths/binomial_distribution.py:33:58: E226 missing whitespace around arithmetic operator
./maths/binomial_distribution.py:52:24: E226 missing whitespace around arithmetic operator
./maths/binomial_distribution.py:56:1: E305 expected 2 blank lines after class or function definition, found 1
6     E226 missing whitespace around arithmetic operator
1     E305 expected 2 blank lines after class or function definition, found 1
7
```

**Comment by cclauss** [comment]
I think we are close to landing this one....  Just a few todos.
1. Edit the first message at the top of this page to do markdown checkboxes correctly
    * https://docs.github.com/en/github/managing-your-work-on-github/about-task-lists#creating-task-lists
2. Let's `from math import factorial` and remove your version
    * https://docs.python.org/3/library/math.html#math.factorial
3. Fix the mypy error that is failing on Travis CI
    * maths/binomial_distribution.py:33: error: Incompatible types in assignment (expression has type "float", variable has type "int")

**Comment by XxSamixX123** [comment]
> succes

I'm quite sure that was fixed earlier.

**Pull request merged**